yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -823,6 +822,14 @@ private[spark] class Client(
     val securityManager = new SecurityManager(sparkConf)
     amContainer.setApplicationACLs(
       YarnSparkHadoopUtil.getApplicationAclsForYarn(securityManager).asJava)
+
+    if (CryptoConf.isShuffleEncryted(sparkConf)) {
+      CryptoConf.initSparkShuffleCredentials(sparkConf, credentials)
+      val dob: DataOutputBuffer = new DataOutputBuffer
```

Comment:
Everything after this line seems unnecessary; that's exactly what `setupSecurityToken` already does.